Sources/Identity/CustomerInfoManager.swift

```diff
@@ -249,7 +249,10 @@ class CustomerInfoManager {
             }
 
             $0.lastSentCustomerInfo = customerInfo
-            self.operationDispatcher.dispatchOnMainThread { [observers = $0.customerInfoObserversByIdentifier] in
+
+            // This must be async to prevent deadlocks if the observer calls a method that ends up reading
+            // this class' data. By making it async, the closure is invoked outside of the lock.
```

Comment:
already asked through slack, just wondering why we need a read lock for customerInfo in the first place

Reply:
Hmm yeah I think that's leftover from the prior implementation. This whole method could be a lot simpler if we just read the underlying data (info, observers, etc) at the beginning and we do the rest outside of the lock, then we wouldn't need the dispatch async change. I'll do that and it should still make the tests pass 👍🏻